src/app/components/champion/ChampionSelector.tsx

```diff
@@ -1,46 +1,66 @@
-import React, {useCallback, useEffect, useState} from "react";
+import React, {useCallback, useState} from "react";
+import axios from 'axios';
 import {Champion, Data} from "../model/models";
 import {ChampionView} from "./ChampionView";
-import {getUrlChampion, getUrlChampionList} from "../../config/config";
-import axios from 'axios';
-import {defaultChampion} from "./default";
-
+import {getUrlChampion, getUrlChampionAvatar, getUrlChampionList} from "../../config/config";
+import {useFetchAPI} from "../../api/reducer";
+import {Autocomplete} from "@material-ui/lab";
+import {Avatar, List, TextField} from "@material-ui/core";
+import {makeStyles} from "@material-ui/core/styles";
 
 export const ChampionSelector = () => {
-    const [champion, setChampion] = useState<Champion>(defaultChampion);
-    const [championsList, setChampionsList] = useState<Champion[]>([]);
+    const [championId, setChampionId] = useState("Aatrox")
+    const championListState = useFetchAPI<Data>({data: {}}, () => axios.get(getUrlChampionList()))
+    const championState = useFetchAPI<Data>(undefined, () => axios.get(getUrlChampion(championId)), [championId])
```

Comment:
Can you not use Data ? It would be better to have an object with types describing your data. I know that what riot api is definitely not good but you can define what it looks like and it would be easier to read yourself later.
I'm 100% sure you can do it.

Reply:
Alright, you cannot describe your type. I agree

Reply:
Yes, thanks to this very well conceived API format !